src/Hosting/Hosting/src/Internal/HostingEventSource.cs

```diff
@@ -51,5 +74,40 @@ public void UnhandledException()
         {
             WriteEvent(5);
         }
+
+        internal void RequestFailed()
+        {
+            Interlocked.Increment(ref _failedRequests);
+        }
+
+        protected override void OnEventCommand(EventCommandEventArgs command)
+        {
+            if (command.Command == EventCommand.Enable)
```

Comment:
What about Disable?

Reply:
From below this line:

> `// They aren't disabled afterwards...`

Do you have more context on why?

Reply:
It's racy to disable counters and dispose and re-create them, that would need to be synchronized. The other way to turn it off is to stop pumping the data (that can be done via a different set of arguments), which turns the timer off so allocating the counter itself is fine.